runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/block/FileBlock.java

```diff
@@ -83,11 +96,13 @@ public FileBlock(final String blockId,
    */
   private void writeToFile(final Iterable<SerializedPartition<K>> serializedPartitions)
     throws IOException {
-    try (FileOutputStream fileOutputStream = new FileOutputStream(filePath, true)) {
+    try (FileChannel fileOutputChannel = new FileOutputStream(filePath, true).getChannel()) {
       for (final SerializedPartition<K> serializedPartition : serializedPartitions) {
         // Reserve a partition write and get the metadata.
         metadata.writePartitionMetadata(serializedPartition.getKey(), serializedPartition.getLength());
-        fileOutputStream.write(serializedPartition.getData(), 0, serializedPartition.getLength());
+        for (final ByteBuffer buffer: serializedPartition.getBuffer()) {
+          fileOutputChannel.write(buffer);
```

Comment:
Why is `getLength()` no longer needed?

Reply:
The previous implementation using the `DirectByteArrayOutputStream` included invalid bytes in the `serializedPartition` so we needed to specify the valid length in the data to only fetch the valid data. Data that is written with `ByteBuffer`s using `DirectByteArrayOutputStream` has a `limit` which automatically stops reading the data when it is not valid even without specifying the length, so we no longer need one.

Reply:
Sounds very good!